From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Tue, 7 Oct 2014 14:49:55 +0200
Subject: [PATCH] media: coda: jpeg: read back thumbnail indication and JPEG
 format

Make sure that thumbnailing is disabled and report the JPEG source
format returned by the SEQ_INIT command.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/media/platform/chips-media/coda-bit.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/platform/chips-media/coda-bit.c b/drivers/media/platform/chips-media/coda-bit.c
index 2736a902e3df..d74d9cc5455f 100644
--- a/drivers/media/platform/chips-media/coda-bit.c
+++ b/drivers/media/platform/chips-media/coda-bit.c
@@ -2018,6 +2018,15 @@ static int __coda_decoder_seq_init(struct coda_ctx *ctx)
 					  (top_bottom & 0x3ff);
 	}
 
+	if (src_fourcc == V4L2_PIX_FMT_JPEG) {
+		val = coda_read(dev, CODA_RET_DEC_SEQ_JPG_THUMB_IND) & 0x1;
+		if (val)
+			v4l2_warn(&dev->v4l2_dev, "thumbnail enabled\n");
+		val = coda_read(dev, CODA_RET_DEC_SEQ_JPG_PARA) & 0x7;
+		coda_dbg(1, ctx, "JPEG source format: %d (%s)\n", val,
+			 val ? "4:2:2" : "4:2:0");
+	}
+
 	if (dev->devtype->product != CODA_DX6) {
 		u8 profile, level;
 
